cli/lib/pm2-manager.ts

```diff
@@ -212,3 +220,70 @@ export async function stopProcess( processName: string ): Promise< void > {
 		} );
 	} );
 }
+
+/**
+ * Subscribe to PM2 process events (online, exit, stop, restart)
+ * @param handler - Callback invoked when a process event occurs
+ * @returns Unsubscribe function to stop listening
+ */
+export async function subscribeProcessEvents(
+	handler: ( data: ProcessEventData ) => void
+): Promise< () => void > {
+	const bus = await getPm2Bus();
+
+	const eventHandler = ( data: unknown ) => {
+		const result = pm2ProcessEventSchema.safeParse( data );
+		if ( ! result.success ) {
+			return;
+		}
+
+		handler( {
+			processName: result.data.process.name,
+			event: result.data.event,
+		} );
+	};
+
+	bus.on( 'process:event', eventHandler );
+
+	return () => {
+		bus.off( 'process:event', eventHandler );
+	};
+}
+
+export interface ProcessMessageData {
+	processName: string;
+	pmId: number;
+	topic: string;
+	data?: unknown;
+}
+
+/**
+ * Subscribe to PM2 process messages (IPC messages from child processes)
+ * @param handler - Callback invoked when a process message is received
+ * @returns Unsubscribe function to stop listening
+ */
+export async function subscribeProcessMessages(
+	handler: ( data: ProcessMessageData ) => void
+): Promise< () => void > {
+	const bus = await getPm2Bus();
+
+	const messageHandler = ( packet: unknown ) => {
+		const result = pm2ProcessMessageSchema.safeParse( packet );
```

Comment:
```suggestion
		const result = childMessagePm2Schema.safeParse( packet );
```
Let's reuse `childMessagePm2Schema` here. We can update the schema definition to also include `data.process.name`